src/Chainweb/BlockHeader.hs

```diff
@@ -345,7 +345,25 @@ isLastInEpoch h = case effectiveWindow h of
         | otherwise -> False
 {-# INLINE isLastInEpoch #-}
 
--- | Compute the POW target for a new BlockHeader
+-- | If it is discovered that the last DA occured significantly in the past, we
+-- assume that a large amount of hash power has suddenly dropped out of the
+-- network. Thus we must perform Emergency Difficulty Adjustment to avoid
+-- stalling the chain.
+--
+slowEpoch :: BlockHeader -> BlockCreationTime -> Bool
+slowEpoch p (BlockCreationTime ct) = actual > (expected * 5)
```

Comment:
This compares against the `BlockCreationTime` of the current `BlockHeader`. The parent is only consulted for the `EpochStartTime`.

While we imagined two scenarios (slow / stalled), I believe this check guards against both of those.